R/lazyframe__lazy.R

```diff
@@ -926,6 +982,37 @@ LazyFrame_dtypes = method_as_property(function() {
     unwrap("in $dtypes()")
 })
 
+#' @title Collect and profile a lazy query.
+#' @description This will run the query and return a tuple containing the materialized DataFrame and
+#'  a DataFrame that contains profiling information of each node that is executed.
+#' @details The units of the timings are microseconds.
+#'
+#' @keywords LazyFrame
+#' @return List of two DataFrames, (collected result, profile stats)
+#' @examples
+#'
+#' #Use $profile() to compare two queries
+#'
+#' # print one '.', take a series convert to r vector, take first value, add 5
+#' r_func = \(s) {cat(".");s$to_r()[1] + 5}
```

Comment:
example hard to grasp, isn't there a most simple / realistic example?

Reply:
adding example `pl$LazyFrame()$select(pl$lit(2) + 2)$profile()`. Ordering other examples by complexity.